azurelinuxagent/common/protocol/wire.py

```diff
@@ -768,6 +799,35 @@ def _update_from_goal_state(self, refresh_type):
 
         raise ProtocolError("Exceeded max retry updating goal state")
 
+    def _save_goal_state_if_necessary(self, new_goal_state, refresh_type):
+        # We cache the goal state if one of the following is true
+        # 1) We didn't previously have a goal state
+        # 2) The incarnation changed
+        # 3) The extensions config changed
+        # 4) We have an update type of GoalStateForced
+        save_goal_state = False
+        if self._goal_state is None or new_goal_state.incarnation != self._goal_state.incarnation:
+            save_goal_state = True
+            self._goal_state = new_goal_state
+        elif refresh_type == WireClient._UpdateType.GoalStateForced:
+            save_goal_state = True
+            self._goal_state = new_goal_state
+        elif new_goal_state.ext_conf is not None:
+            if new_goal_state.ext_conf.changed:
```

Comment:
on this path we update _goal_state, but we do not save it disk... why is that?

Reply:
Logic has changed